spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/trace/sleuth/StackdriverTraceSpanListenerTests.java

```diff
@@ -47,8 +47,9 @@
 import org.springframework.mock.env.MockEnvironment;
 import org.springframework.test.context.junit4.SpringRunner;
 
-@SpringBootTest(classes = {
-		StackdriverTraceSpanListenerTests.TestConfiguration.class })
+@SpringBootTest(
```

Comment:
@artembilan @snicoll @dsyer these tests are activating `GcpCloudSqlAutoConfiguration` for some reason, even though it's not listed in `classes`.
The other "sibling" autoconfig test, `StackdriverTraceAutoConfigurationTests` isn't enabling `GcpCloudSqlAutoConfiguration`.
I can't figure out what the difference is. Would you know off the top of your head?

Reply:
@joaoandremartins ,

look, in the end of this class you have a configuration class:
```
@Configuration
@EnableAutoConfiguration
static class TestConfiguration {
```
So, that `@EnableAutoConfiguration` definitely triggers all the possible auto-configurations to be loaded.

@snicoll , suggested once to follow the Spring Boot way to test your requirements. Like load only those configurations you need for this particular test. For example I did this in one of my tests: https://github.com/spring-projects/spring-integration-java-dsl/blob/master/src/test/java/org/springframework/integration/dsl/test/jpa/JpaTests.java#L170

Reply:
Thanks a lot for the tip @artembilan !

> @snicoll , suggested once to follow the Spring Boot way to test your requirements. Like load only those configurations you need for this particular test. For example I did this in one of my tests: https://github.com/spring-projects/spring-integration-java-dsl/blob/master/src/test/java/org/springframework/integration/dsl/test/jpa/JpaTests.java#L170

That's already in scope for #313 . Let's take care of all of them in a single stroke.